pkg/kubeconfig/kubeconfig.go

```diff
@@ -47,13 +47,13 @@ func NewKubeconfig(coreClient kubernetes.Interface, log logr.Logger) *Kubeconfig
 
 // ClusterAccess takes cluster info and a ServiceAccount Name, and returns a populated kubeconfig that can connect to a cluster.
 // if the saName is empty then you'll connect to a cluster via the clusterOpts inside the genericOpts, otherwise you'll use the specified SA.
-func (k Kubeconfig) ClusterAccess(saName string, clusterOpts *v1alpha1.AppCluster, accessLocation AccessLocation) (AccessInfo, error) {
+func (k Kubeconfig) ClusterAccess(saName string, clusterOpts *v1alpha1.AppCluster, accessLocation AccessLocation, preferredNamespace string) (AccessInfo, error) {
 	var err error
 	var clusterAccessInfo AccessInfo
 
 	switch {
 	case len(saName) > 0:
-		clusterAccessInfo, err = k.serviceAccounts.Find(accessLocation, saName)
+		clusterAccessInfo, err = k.serviceAccounts.Find(accessLocation, saName, preferredNamespace)
```

Comment:
I am thinking that it makes sense to add the preferredNamespace to clusterAccessInfo for both service accounts and clusterOpts.